src/analyze/analyze-chid.c

```diff
@@ -243,6 +348,30 @@ int verb_chid(int argc, char *argv[], void *userdata) {
         if (r < 0)
                 return r;
 
+        if (arg_drm_card_path) {
+                _cleanup_(sd_device_unrefp) sd_device *drm_card = NULL;
+                r = sd_device_new_from_path(&drm_card, arg_drm_card_path);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to open device %s: %m", arg_drm_card_path);
+
+                const char *subsys = NULL;
+                r = sd_device_get_subsystem(drm_card, &subsys);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to get subsystem for device %s: %m", arg_drm_card_path);
+                if (!streq(subsys, "drm"))
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot read EDID from a \"%s\" device", subsys);
+
+                r = edid_parse(drm_card, &smbios_fields[CHID_EDID_PANEL]);
+                if (r < 0)
+                    return log_error_errno(r, "Failed to parse EDID for device %s: %m", arg_drm_card_path);
+        } else {
+                r = edid_search(&smbios_fields[CHID_EDID_PANEL]);
+                if (r == -ENOTUNIQ)
+                        return r;
+                else if (r != -ENODEV && r < 0)
```

Comment:
the `else` is redundan, given that the first branch definitely returns already. please drop it, we like short code, i.e. redundant words removed.